app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala

```diff
@@ -312,13 +313,20 @@ class BusinessEntityController @Inject()(
     mapping("removeBusiness" -> mandatory("removeBusiness"))(FormValues.apply)(FormValues.unapply))
 
   lazy val eoriForm: Form[FormValues] = Form(
-    mapping("businessEntityEori" -> mandatory("businessEntityEori"))(eoriEntered => FormValues(s"$eoriPrefix$eoriEntered"))(eori => eori.value.drop(2).some).verifying(
-    "businessEntityEori.regex.error",
-    fields => fields match {
-      case a if a.value.matches(EORI.regex) => true
-      case _ => false
-    }
-  ))
+    mapping("businessEntityEori" -> mandatory("businessEntityEori"))(eoriEntered => FormValues(s"$eoriPrefix$eoriEntered"))(eori => eori.value.drop(2).some)
+    .verifying("businessEntityEori.error.incorrect-length",
+      fields => fields match {
+        case a if a.value.toString().length == 14 || a.value.toString().length == 17 => true
```

Comment:
out of curosity why we are checking for 14 and 17 , isn't the eoRI length should be 12 or 15 including the GB part?

Reply:
Excluding the GB part is what we understand.

Reply:
but then   it contradicts this, isn't?

```
  type EORI = String @@ EORI.Tag
  object EORI extends RegexValidatedString(
    """^(GB|XI)[0-9]{12,15}$"""
```

Reply:
Nahhh it's a bit tricky to read but that means...
"(GB or XI) followed by (a member of the alphabet 0-9 (syntax sugar for digits) 12 or 15 times)"

Reply:
E.e

`scala> """^(GB|XI)[0-9]{12,15}$""".r.matches("GB1234567890")
val res1: Boolean = false

scala> """^(GB|XI)[0-9]{12,15}$""".r.matches("GB123456789012")
val res2: Boolean = true

scala> """^(GB|XI)[0-9]{12,15}$""".r.matches("GB123456789012345")
val res3: Boolean = true`

Reply:
yup got it as I was typing  🙈